src/geode/model/helpers/convert_brep_section.cpp

```diff
@@ -59,10 +68,310 @@ namespace
         BuilderTo& builder_to,
         const geode::ModelCopyMapping& mappings )
     {
-        builder_to.create_unique_vertices( from.nb_unique_vertices() );
+        const auto first_new_unique_vertex_id =
+            builder_to.create_unique_vertices( from.nb_unique_vertices() );
         geode::detail::copy_vertex_identifier_components(
-            from, builder_to, mappings );
+            from, builder_to, first_new_unique_vertex_id, mappings );
     }
+
+    class SectionExtruder
+    {
+    public:
+        SectionExtruder( const geode::Section& section,
+            geode::index_t axis_to_extrude,
+            double min_coordinate,
+            double max_coordinate )
+            : section_{ section },
+              brep_{},
+              brep_builder_{ brep_ },
+              section_slice0_mapping_{},
+              slice0_brep_mapping_{},
+              section_slice1_mapping_{},
+              slice1_brep_mapping_{}
+        {
+            auto slice0_conversion_output = convert_section_into_brep(
+                section_, axis_to_extrude, min_coordinate );
+            section_slice0_mapping_ = std::get< 1 >( slice0_conversion_output );
+            slice0_brep_mapping_ =
+                brep_builder_.copy( std::get< 0 >( slice0_conversion_output ) );
+
+            auto slice1_conversion_output = convert_section_into_brep(
+                section_, axis_to_extrude, max_coordinate );
+            section_slice1_mapping_ = std::get< 1 >( slice1_conversion_output );
+            geode::BRepConcatener brep_concatener( brep_ );
+            slice1_brep_mapping_ = brep_concatener.concatenate(
+                std::get< 0 >( slice1_conversion_output ) );
+        }
+
+        ~SectionExtruder() = default;
+        geode::BRep extrude()
+        {
+            create_lines();
+            create_surfaces();
+            create_blocks();
+            return std::move( brep_ );
+        }
+
+    private:
+        geode::uuid section_slice0_brep_mapping(
+            const geode::ComponentType& brep_component_type,
+            geode::uuid section_component )
+        {
+            return (
+                slice0_brep_mapping_.at( brep_component_type )
+                    .in2out( section_slice0_mapping_.at( brep_component_type )
+                                 .in2out( section_component ) ) );
```

Comment:
```suggestion
            return 
                slice0_brep_mapping_.at( brep_component_type )
                    .in2out( section_slice0_mapping_.at( brep_component_type )
                                 .in2out( section_component ) );
```
This is not needed and verbose

Reply:
Replace for all the file